packages/definitions/dist/fhir/r4/profiles-medplum.json

```diff
@@ -3849,5 +3849,356 @@
+            "id" : "Agent.setting.value[x]",
+            "path" : "Agent.setting.value[x]",
+            "definition" : "The setting value.",
+            "min" : 1,
+            "max" : "1",
+            "type" : [{
+              "code" : "string"
+            },
+            {
+              "code" : "boolean"
+            },
+            {
+              "code" : "decimal"
+            },
+            {
+              "code" : "integer"
+            }],
+            "base" : {
+              "path" : "Agent.setting.value[x]",
+              "min" : 1,
+              "max" : "1"
+            }
+          },
+          {
+            "id" : "Agent.channel",
+            "path" : "Agent.channel",
+            "short" : "The channel on which to report matches to the criteria",
+            "definition" : "Details where to send notifications when resources are received that meet the criteria.",
+            "min" : 1,
+            "max" : "*",
+            "base" : {
+              "path" : "Agent.channel",
+              "min" : 1,
+              "max" : "*"
+            },
+            "type" : [{
+              "code" : "BackboneElement"
+            }]
+          },
+          {
+            "id" : "Agent.channel.name",
+            "path" : "Agent.channel.name",
+            "short" : "The human readable friendly name of the channel.",
+            "definition" : "The human readable friendly name of the channel.",
+            "min" : 1,
+            "max" : "1",
+            "base" : {
+              "path" : "Agent.channel.name",
+              "min" : 1,
+              "max" : "1"
+            },
+            "type" : [{
+              "code" : "string"
+            }]
+          },
+          {
+            "id" : "Agent.channel.status",
+            "path" : "Agent.status",
+            "short" : "active | off | error",
+            "definition" : "The status of the channel.",
+            "min" : 1,
+            "max" : "1",
+            "base" : {
+              "path" : "Agent.channel.status",
+              "min" : 1,
+              "max" : "1"
+            },
+            "type" : [{
+              "code" : "code"
+            }],
+            "binding" : {
+              "strength" : "required",
+              "valueSet" : "https://medplum.com/fhir/ValueSet/agent-status|4.0.1"
+            }
+          },
+          {
+            "id" : "Agent.channel.device",
+            "path" : "Agent.channel.device",
+            "definition" : "Optional device resource representing the device connected to the channel.",
+            "min" : 0,
+            "max" : "1",
+            "type" : [{
+              "code" : "Reference",
+              "targetProfile" : ["http://hl7.org/fhir/StructureDefinition/Device"]
+            }],
+            "base" : {
+              "path" : "Agent.channel.device",
+              "min" : 0,
+              "max" : "1"
+            }
+          },
+          {
+            "id" : "Agent.channel.target",
```

Comment:
Instead of assuming this will always be a Bot, we might want to consider reusing the BackboneElement `SubscriptionChannel` (i.e. [Subscription.channel](https://hl7.org/fhir/R4/subscription.html)). That would at least allow us to send to arbitrary endpoints fi the use case arises.

Reply:
I want to make the common case more obvious.  `Subscription.channel.endpoint` is a string, and then you need to know about magic `Bot/{id}` values, you don't get autocompletes, etc.

I'm open to considering `target[x]` choice-of-type, so you could do `targetUrl` to a URL or `targetReference` to a Bot or other resource.

Reply:
I also want to be wary of blowing up the scope here - opening up `target` to arbitrary endpoints is not a requirement, and we do not have any plans of implementing it in the near future.